### ISSUE TITLE
Handle EC2 Spot Termination Notice

### ISSUE THREAD

**Issue by hjacobs** [zalando-incubator/kubernetes-on-aws#1238]
We should automatically drain nodes which receive an EC2 Spot Termination Notice. Example for ECS (from [AWS blog post](https://aws.amazon.com/blogs/compute/powering-your-amazon-ecs-cluster-with-amazon-ec2-spot-instances/)):

```bash
#!/bin/bash

while sleep 5; do
  if [ -z $(curl -Isf http://169.254.169.254/latest/meta-data/spot/termination-time) ]; then
    /bin/false
  else
    # TODO: do kubectl drain instead
    ECS_CLUSTER=$(curl -s http://localhost:51678/v1/metadata | jq .Cluster | tr -d \")
    CONTAINER_INSTANCE=$(curl -s http://localhost:51678/v1/metadata \
      | jq .ContainerInstanceArn | tr -d \")
    aws ecs update-container-instances-state --cluster $ECS_CLUSTER \
      --container-instances $CONTAINER_INSTANCE --status DRAINING
  fi
done

**Comment by mikkeloscar** [comment]
Just a note. This is only a 2 minute warning so it's limited how much draining we can do. Still it's better than nothing but it will affect applications that require longer time to move between nodes (e.g. postgres cluster pods) and it should be reflected in our SLAs.

**Comment by hjacobs** [comment]
There are multiple existing Spot termination handlers for Kubernetes, we should use one of them:

* https://github.com/kube-aws/kube-spot-termination-notice-handler
* https://github.com/pusher/k8s-spot-termination-handler

**Comment by aermakov-zalando** [comment]
Fixed in #1296.
